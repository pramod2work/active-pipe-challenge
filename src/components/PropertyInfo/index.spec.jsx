import React from 'react'
import { shallow, mount } from 'enzyme'

import { PropertyInfo } from './index'

describe('PropertyInfo Component test', () => {
  test('Component Snapshot Testing', () => {
    const props = {
      "id": 1,
      "status": "current",
      "street": "1 Fake Street",
      "suburb": "Glen Iris",
      "state": "VIC",
      "postcode": "3146",
      "price": 1230500,
      "image": "https://code-challenge.activepipe.com/property-images/frontage/01.jpg"
    }
    const component = shallow(<PropertyInfo {...props} />)
    expect(component).toMatchSnapshot()
  })

  test('Component Snapshot Testing for invalid status', () => {
    const props = {
      "id": 1,
      "status": "not_listed",
      "street": "1 Fake Street",
      "suburb": "Glen Iris",
      "state": "VIC",
      "postcode": "3146",
      "price": 1230500,
      "image": "https://code-challenge.activepipe.com/property-images/frontage/01.jpg"
    }
    const component = mount(<PropertyInfo {...props} />)
    expect(component).toMatchSnapshot()
  })
})