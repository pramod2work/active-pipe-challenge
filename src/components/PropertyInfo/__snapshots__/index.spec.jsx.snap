// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyInfo Component test Component Snapshot Testing 1`] = `
<div
  className="cell"
>
  <div>
    <span
      className="currency"
    >
      $
      1,230,500
    </span>
    <Status
      status="current"
    />
  </div>
  <img
    alt="property"
    className="propertyImg"
    src="https://code-challenge.activepipe.com/property-images/frontage/01.jpg"
  />
  <div
    className="address"
  >
    <div>
      1 Fake Street
    </div>
    <div>
      Glen Iris
    </div>
    <div>
      VIC
       
      3146
    </div>
  </div>
</div>
`;

exports[`PropertyInfo Component test Component Snapshot Testing for invalid status 1`] = `
<PropertyInfo
  id={1}
  image="https://code-challenge.activepipe.com/property-images/frontage/01.jpg"
  postcode="3146"
  price={1230500}
  state="VIC"
  status="not_listed"
  street="1 Fake Street"
  suburb="Glen Iris"
>
  <div
    className="cell"
  >
    <div>
      <span
        className="currency"
      >
        $
        1,230,500
      </span>
      <Status
        status="not_listed"
      >
        <span
          className="status"
        >
          <span>
            not_listed
          </span>
          <span
            className="not_listed"
          />
        </span>
      </Status>
    </div>
    <img
      alt="property"
      className="propertyImg"
      src="https://code-challenge.activepipe.com/property-images/frontage/01.jpg"
    />
    <div
      className="address"
    >
      <div>
        1 Fake Street
      </div>
      <div>
        Glen Iris
      </div>
      <div>
        VIC
         
        3146
      </div>
    </div>
  </div>
</PropertyInfo>
`;
